lib/ssh_scan/rule_engine.rb

```diff
@@ -0,0 +1,30 @@
+module SSHScan
+  class RuleEngine
+    def self.match(rules, result)
+      answer = true
+      rules.each do |key, value|
+        conditional = key.split("_")[-1]
+        attribute = key.chomp("_" + conditional).to_sym
+        if result[attribute]
+          if conditional == "eq"
+            answer &= (result[attribute] == value)
+          elsif conditional == "neq"
+            answer &= (result[attribute] != value)
+          elsif result[attribute].is_a? Numeric
+            if conditional == "gt"
+              answer &= (result[attribute] > value)
+            elsif conditional == "lt"
+              answer &= (result[attribute] < value)
+            elsif conditional == "gte"
+              answer &= (result[attribute] >= value)
+            elsif conditional == "lte"
+              answer &= (result[attribute] <= value)
+            end
+          end
+        end
```

Comment:
This block of nested if's might be a better fit as a case statement or it's own unique object class that can cleanly parse a single rule.